src/toolkit/pyroprocessing/pyre_refining.cc

```diff
@@ -0,0 +1,185 @@
+#include "pyre_refining.h"
+#include "boost/bind.hpp"
+#include <boost/math/tools/roots.hpp>
+
+
+using cyclus::Material;
+using cyclus::Composition;
+using cyclus::CompMap;
+using cyclus::ValueError;
+
+using boost::math::tools::bisect;
+
+namespace pyro {
+
+Refine::Refine() {
+  temp(0);
+  pressure(0);
+  rotation(0);
+  b_size(0);
+  Rtime(0);
+}
+
+Refine::Refine(double new_temp = 900, 
+               double new_press = 760, 
+               double new_rotation = 0, 
+               double new_batch_size = 20,
+               double new_rtime = 1
+            ) 
+            {
+  SetCoeff();
+  temp(new_temp);
+  pressure(new_press);
+  rotation(new_rotation);
+  b_size(new_batch_size);
+  Rtime(new_rtime);
+}
+
+void Refine::SetCoeff() {
+  t0_ = 4.7369E-9;
+  t1_ = -1.08337E-5;
+  t2_ = 0.008069;
+  t3_ = -0.9726;
+  p0_ = -7.17631E-10;
+  p1_ = 4.04545E-07;
+  p2_ = -8.06336E-05;
+  p3_ = 1.002;
+  a0_ = 0.032;
+  a1_ = 0.72;
+  a2_ = 0.0338396;
+  a3_ = 0.83667;
+}
+
+struct Refine::TerminationCondition {
+    bool operator() (double min, double max)  {
+        return abs(min - max) <= 0.000001;
```

Comment:
why `0.00001` and not `0.0000001`

also maybe `1e-5` would be more readable...